sdk/eventhub/Azure.Messaging.EventHubs/stress/src/Program.cs

```diff
@@ -13,215 +13,358 @@
+            if (string.IsNullOrEmpty(roleString))
+            {
+                foreach (var role in roleList)
                 {
-                    concurrentBufferedProducerTestConfig.EventHub = _promptForResources("Event Hub name", "concurrent buffered producer test", concurrentBufferedProducerTestConfig.EventHub);
+                    scenarioTasks.Add(_runRole(role, testConfiguration, roleConfiguration, metrics, cancellationToken));
                 }
+            }
+            else
+            {
+                scenarioTasks.Add(_runRole(roleList[int.Parse(roleString)], testConfiguration, roleConfiguration, metrics, cancellationToken));
+            }
 
-                concurrentBufferedProducerTestConfig.ConcurrentSends = 5;
+            while (!cancellationToken.IsCancellationRequested)
+            {
+                _updateEnvironmentStatistics(metrics);
 
-                var testRun = new BufferedProducerTest(concurrentBufferedProducerTestConfig);
-                testRunTasks.Add(testRun.Run());
+                await Task.Delay(TimeSpan.FromMinutes(5), cancellationToken).ConfigureAwait(false);
             }
 
-            // Wait for all test runs to complete
+            await Task.WhenAll(scenarioTasks).ConfigureAwait(false);
+        }
+        catch (TaskCanceledException)
+        {
+            // Run is complete
+        }
+        catch (Exception ex) when
+            (ex is OutOfMemoryException
+            || ex is StackOverflowException
+            || ex is ThreadAbortException)
+        {
+            throw;
+        }
+        catch (Exception ex)
+        {
+            metrics.Client.TrackException(ex);
+        }
+        finally
+        {
+            metrics.Client.Flush();
+            await Task.Delay(60000).ConfigureAwait(false);
+        }
+    }
+
+    /// <summary>
+    ///   Starts a background task for each test that needs to be run, and waits for all
+    ///   test runs to completed before returning.
+    /// </summary>
+    ///
+    /// <param name="roleList">The list of roles needed for the scenario being run.</param>
+    /// <param name="testConfiguration">The <see cref="TestConfiguration" /> instance used to configure this test scenario run.</param>
+    /// <param name="roleConfiguration">The <see cref="RoleConfiguration" /> instance used to configure the role being run.</param>
+    /// <param name="metrics">The <see cref="Metrics" /> instance used by this role to send metrics to application insights.</param>
+    /// <param name="jobIndex">If this is a single-role run, this is the value used to determine which role should be run.</param>
+    /// <param name="cancellationToken">The <see cref="CancellationToke"/> instance to signal the request to cancel the operation.</param>
+    ///
+    private static async Task _runRole(string role, TestConfiguration testConfiguration, RoleConfiguration roleConfiguration, Metrics metrics, CancellationToken cancellationToken)
```

Comment:
nit:  If we're splitting the params of the other methods onto separate lines, we should consider doing the same here.